coalib/parsing/FilterHelper.py

```diff
@@ -123,10 +126,20 @@ def collect_filters(args, arg_list=None, arg_parser=None):
         Instance of ArgParser that is used to parse arg list.
     :return:
         List of filters in standard filter format, i.e
-        ``[['filter_name', 'arg1', 'arg2']]``.
+        ``{'filter_name': ['arg1', 'arg2']}``.
     """
     if args is None:
         arg_parser = default_arg_parser() if arg_parser is None else arg_parser
         args = arg_parser.parse_args(arg_list)
 
-    return getattr(args, 'filter_by', None) or []
+    filters = getattr(args, 'filter_by', None) or []
+    filters = list_to_dict(filters)
+    return filters
+
+
+def list_to_dict(filters):
```

Comment:
Please name this appropriately, list_to_dict does not explain its purpose.

Reply:
I asked a maintainer regarding what name should I give to this function and he suggested this. Please suggest a better name.

Reply:
`flat_filter_to_dict` or `filter_vector_to_dict` maybe? This is not a very generic function hence that is not a very appropriate name.

Reply:
And have you discussed anything about the tuple/list situation concerning to args?

Reply:
I haven't yet. I'll try to make it work with tuples.